src/test/java/com/puppycrawl/tools/checkstyle/utils/TokenUtilTest.java

```diff
@@ -395,4 +395,79 @@ public void testIsBooleanLiteralType() {
                 .isFalse();
     }
 
+    @Test
+    public void testAstHasModifiers() {
+        final DetailAstImpl astFinal = new DetailAstImpl();
```

Comment:
We cover new code by real test input only, we are not extending our tech debt by allowing coverage by synthetic tests.